etna/core/middleware.py

```diff
@@ -122,3 +125,53 @@ def process_template_response(
             ),
         )
         return response
+
+
+class SearchMiddleware:
+    """
+    Handles the session settings for "back to search results" functionality in the record details page.
+    """
+
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        response = self.get_response(request)
+
+        return response
+
+    def process_view(self, request, view_func, view_args, view_kwargs):
+        # create or retain session var
+        if hasattr(view_func, "view_class"):
+            # set session when navigating from search view containing links to record details page
+            if view_func.view_class in (CatalogueSearchView, FeaturedSearchView):
+                request.session.set_expiry(settings.SESSION_EXPIRY_VALUE)
```

Comment:
Is the aim here to limit the length of time the value is respected? Why do we care?

Reply:
Setting the session to expire here affects more than just the life of this data (users will be logged out earlier too). We should think of alternative solutions really.

Reply:
> Is the aim here to limit the length of time the value is respected? Why do we care?
You are right. I checked without the settings its the same, not required.

> Setting the session to expire here affects more than just the life of this data (users will be logged out earlier too). We should think of alternative solutions really.
I don't know another solution. Please advise.

Reply:
As far as I can tell, you have two options:
1. Don't worry about it: The session will expire when it expires, and the data will no longer be available.
2. Add a timestamp to the session at the same time, and use that to decide how long to respect the session data for.

Reply:
>2. Add a timestamp to the session at the same time, and use that to decide how long to respect the url value for.

I'm a bit confused with point 2) with 
> Setting the session to expire here affects more than just the life of this data (users will be logged out earlier too).

Wouldn't the user get logged out on the timestamp?

Reply:
I'm not saying to set an expiry on the session... just to add a timestamp to the data, like we are doing for the URL value.

Reply:
Thanks Andy, now updated for option 2)